lifx-core/src/lib.rs

```diff
@@ -50,33 +51,21 @@ impl std::convert::From<io::Error> for Error {
     }
 }
 
+impl From<std::convert::Infallible> for Error {
+    fn from(_: std::convert::Infallible) -> Self {
+        unreachable!()
+    }
+}
+
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "An error occurred.")
     }
 }
 
-trait LifxFrom<T>: Sized {
-    fn from(val: T) -> Result<Self, Error>;
-}
-
-macro_rules! derive_lifx_from {
-{ $( $t:ty ),*} => {
-    $(
-        impl LifxFrom<$t> for $t {
-            fn from(val: $t) -> Result<Self, Error> { Ok(val)}
-        }
-    )*
-
-}
-}
-
-derive_lifx_from! {
-    u8, u16, i16, u32, f32, u64, LifxIdent, LifxString, EchoPayload, HSBK
-}
-
-impl LifxFrom<u8> for ApplicationRequest {
-    fn from(val: u8) -> Result<ApplicationRequest, Error> {
+impl TryFrom<u8> for ApplicationRequest {
```

Comment:
For fun, I looked at this history of this file.  I introduced the `LifxFrom` trait back in August 2018, which is about 8 months before `TryFrom` was stabilized 😄   So this is a nice modernization!